content/blog/2018-06-26-roomba.Rmd

```diff
@@ -0,0 +1,107 @@
+
+
+## The birth of roomba
+
+
+It was clear Amanda was on to something good by the lively discussion in the [#runconf18](https://github.com/ropensci/unconf18/issues/) issues repository leading up to the unconf. Thanks to input from Jenny Bryan, Jim Hester, Carl Boettinger, Scott Chamberlain, Bob Rudis, and Noam Ross, we had a lot of ideas to work with when the unconf began. Fortunately, Jim already had a function called `dfs_idx()` ([here](https://github.com/ropenscilabs/roomba/blob/master/R/dfs_idx.R)) written to perform depth-first searches of nested lists from the [GitNub GraphQL API](https://developer.github.com/v4/). With the core list-traversal code out of the way, we split our efforts between developing a usable interface, stockpiling `.JSON` files to test on, and developing a Shiny app.
+
+
+## What's working
+We've got the basic structure of `roomba` sorted out, and you should install it from GitHub to try out! Here are a few of the examples we've put together. 
+
+
+```{r}
+library(roomba)
+#load twitter data example
+data(twitter_data)
+
+#roomba-fy!
+roomba(twitter_data, c("created_at", "name"))
+```
+
+And just the *first* element of the `twitter_data` list will show you that `roomba` has simplified this process quite a bit.
+
+```{r}
+twitter_data[[1]]
+
+```
+
+We created a Shiny app too, which in its current state allows you to select a `.Rda` or `.JSON` file, pick two variables, and create a scatterplot of them.
+
+
+Run the app like this:
+```{r eval=FALSE}
+shiny_roomba()
+```
+
+## What's not
+Of course, in two days we weren't able to build a magical one-size-fits-all solution to every API response data headache. Right now, the main barrier to usability is that both the `roomba()` function and `shiny_roomba()` app only work on sub-list items of the same length and same data type stored at the same depth. To illustrate on the `twitter_data`:
+
+```{r}
+#This doesn't work because "user" has data of different types and lengths
+roomba(twitter_data, c("user"))
+
+
+#This doesn't work because "name" and "retweet_count" are at different depths.
+roomba(twitter_data, c("name","retweet_count"))
+```
+
+
+In addition, we've got some features we want to add, such as handling a larger variety of column names (i.e. passing a string for a single column name, keeping all values even if they are all `NULL`). We would love your feedback on other things we can add!
+
+
+## The team 
+
+**Amanda Dobbyn** <br>
+Job: Data Scientist at Earlybird Software <br>
+Project contributions: initial GH issue, package name, wrapper for `dfs_idx()` <br>
+
+**Jim Hester** <br>
+Job: Software Engineer at RStudio <br>
+Project contributions: `dfs_idx()` and `remove_nulls()` functions, package building, README, and debugging <br>
+
+**Christine Stawitz** <br>
+Job: Postdoctoral researcher at UW's School of Aquatic and Fishery Sciences <br>

```

Comment:
spell out "UW", especially since we have an international audience

Reply:
done!